service/dap/server.go

```diff
@@ -1037,16 +1107,65 @@ func (s *Session) onLaunchRequest(request *dap.LaunchRequest) {
 
 		// Start the program on a different goroutine, so we can listen for disconnect request.
 		go func() {
+			if redirected {
+				s.preTerminatedWG.Add(1)
+				go func() {
+					defer s.preTerminatedWG.Done()
+					readerFunc(s.stdoutReader, "stdout")
+				}()
+
+				s.preTerminatedWG.Add(1)
+				go func() {
+					defer s.preTerminatedWG.Done()
+					readerFunc(s.stderrReader, "stderr")
+				}()
+				// Wait for the input and output to be read
+			}
 			if err := cmd.Wait(); err != nil {
 				s.config.log.Debugf("program exited with error: %v", err)
 			}
+
 			close(s.noDebugProcess.exited)
 			s.logToConsole(proc.ErrProcessExited{Pid: cmd.ProcessState.Pid(), Status: cmd.ProcessState.ExitCode()}.Error())
 			s.send(&dap.TerminatedEvent{Event: *newEvent("terminated")})
 		}()
 		return
 	}
 
+	if redirected {
+		redirects, err := NewRedirector()
+		if err != nil {
+			s.sendShowUserErrorResponse(request.Request, InternalError, "Internal Error",
+				fmt.Sprintf("failed to generate stdio pipes - %v", err))
+			return
+		}
+
+		s.config.Debugger.Redirect = redirects
+		s.preTerminatedWG.Add(1)
```

Comment:
Don't you think it's weird that the nodebug code and the debug code here do more or less the same thing, but it's not actually the same code, it's a slightly different copy instead?